src/jvm/main/org/jetbrains/kotlinx/lincheck/ParameterGenerators.kt

```diff
@@ -37,6 +37,44 @@ internal object DummyParameterGenerator : ParameterGenerator<Any?> {
     override fun generate() = throw UnsupportedOperationException()
 }
 
+/**
+ * @param configuration configuration in format EnumType.ConstantA,EnumType.ConstantB,..,EnumType.ConstantN
+ */
+class EnumGen<T : Enum<T>>(enumClass: Class<out T>, randomProvider: RandomProvider, configuration: String) :
+    ParameterGenerator<T> {
+
+    private val enumValues: MutableList<T>
+    private val intGenerator: ExpandingRangeIntGenerator
+    private val shuffleRandom = randomProvider.createRandom()
+
+    init {
+        val allEnumValues = enumClass.enumConstants.toList()
+
+        enumValues = if (configuration.isEmpty()) allEnumValues else {
+            configuration.replace("\\s", "").split(",").map { enumStr ->
```

Comment:
It's safer to replace "\\s*,\\s*'" with "," (not sure that the regex syntax is correct here)

Reply:
We use current approach in all parameter generation parsing methods, why do you think that it's safer?

Reply:
Consider you have an `enum class` in Kotlin with whitespaces in value names. Unfortunately, you cannot specify such values with the current approach, since you remove all whitespaces. Please add the corresponding test and check.